sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle

```diff
@@ -90,75 +97,115 @@ if (gradleInjectionEnabled != "true") {
             def resolutionResult = incoming.resolutionResult
 
-            if (gePluginVersion) {
+            if (develocityPluginVersion) {
                 def scanPluginComponent = resolutionResult.allComponents.find {
-                    it.moduleVersion.with { group == "com.gradle" && (name == "build-scan-plugin" || name == "gradle-enterprise-gradle-plugin") }
+                    it.moduleVersion.with { group == "com.gradle" && ['build-scan-plugin', 'gradle-enterprise-gradle-plugin', 'develocity-gradle-plugin'].contains(name) }
                 }
                 if (!scanPluginComponent) {
-                    logger.lifecycle("Applying $BUILD_SCAN_PLUGIN_CLASS via init script")
-                    applyPluginExternally(pluginManager, BUILD_SCAN_PLUGIN_CLASS)
-                    if (geUrl) {
-                        logger.lifecycle("Connection to Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer, captureFileFingerprints: $geCaptureFileFingerprints")
-                        buildScan.server = geUrl
-                        buildScan.allowUntrustedServer = geAllowUntrustedServer
+                    def pluginClass = dvOrGe(DEVELOCITY_PLUGIN_CLASS, BUILD_SCAN_PLUGIN_CLASS)
+                    logger.lifecycle("Applying $pluginClass via init script")
+                    applyPluginExternally(pluginManager, pluginClass)
+                    def rootExtension = dvOrGe(
+                            { develocity },
+                            { buildScan }
+                    )
+                    def buildScanExtension = dvOrGe(
+                            { rootExtension.buildScan },
+                            { rootExtension }
+                    )
+                    if (develocityUrl) {
+                        logger.lifecycle("Connection to Develocity: $develocityUrl, allowUntrustedServer: $develocityAllowUntrustedServer, captureFileFingerprints: $develocityCaptureFileFingerprints")
+                        rootExtension.server = develocityUrl
+                        rootExtension.allowUntrustedServer = develocityAllowUntrustedServer
                     }
-                    buildScan.publishAlways()
-                    if (isAtLeast(gePluginVersion, '2.1') && atLeastGradle5) {
-                        if (isAtLeast(gePluginVersion, '3.7')) {
-                            buildScan.capture.taskInputFiles = geCaptureFileFingerprints
+                    if (!shouldApplyDevelocityPlugin) {
+                        // Develocity plugin publishes scans by default
+                        buildScanExtension.publishAlways()
+                    }
+                    if (isAtLeast(develocityPluginVersion, '2.1') && atLeastGradle5) {
+                        if (isAtLeast(develocityPluginVersion, '3.17')) {
+                            buildScanExtension.capture.fileFingerprints.set(develocityCaptureFileFingerprints)
```

Comment:
Can you use `=`?

```suggestion
                            buildScanExtension.capture.fileFingerprints = develocityCaptureFileFingerprints
```

Reply:
No, you can't. The assignment operator for properties is supported only from Gradle 8.2

Reply:
That's true for Kotlin DSL, but the Groovy DSL has supported it for much longer.

What is the minimum supported version of Gradle for the Develocity plugin?

Reply:
Right, I am all over the place 🙈 It is Gradle 5.0

Reply:
In that case I would expect that you could use `=`, no?

Reply:
> I am all over the place

That's alright. That's why we have code reviews. 🤣

Reply:
Changed in https://github.com/gradle/actions/pull/62/commits/e35f7f7c74f29dd9df8aa5e1418e759d71f56702

Reply:
You missed this one. 😅

Reply:
Now fixed